src/vl-map-search.js

```diff
@@ -128,91 +128,114 @@ export class VlMapSearch extends vlElement(HTMLElement) {
-            });
+            this._searchChoicesByValue(event.detail.value);
           } else {
-            fetch(this._findLocationsByLambertCoordinaatUrl(lambertCoordinaat)).then((response) => {
-              return response.json();
-            }).then((data) => {
-              const choices = [{
-                value: lambertCoordinaat,
-                label: 'Lambert coördinaat: ' + lambertCoordinaat.toString(),
-              }];
-
-              if (data && data.LocationResult) {
-                data.LocationResult.map((locationResult) => {
-                  return {
-                    value: locationResult,
-                    label: locationResult.FormattedAddress,
-                  };
-                }).forEach((choice) => {
-                  choices.push(choice);
-                });
-              }
-
-              this._selectElement.choices = choices;
-            });
+            this._searchChoicesByLambertCoördinaat(lambertCoordinaat);
           }
         }
       });
     }
   }
 
+  _searchChoicesByValue(searchValue) {
+    fetch(this.searchUrl + searchValue).then((response) => {
+      return response.json();
+    }).then((data) => {
+      if (data && data.SuggestionResult) {
+        const resultaten = data.SuggestionResult.map((resultaat) => {
+          return {
+            value: resultaat,
+            label: resultaat,
+          };
+        });
+        this._selectElement.choices = resultaten;
+      }
+    });
+  }
+
+  _searchChoicesByLambertCoördinaat(lambertCoordinaat) {
+    fetch(this._findLocationsByLambertCoordinaatUrl(lambertCoordinaat)).then((response) => {
+      return response.json();
+    }).then((data) => {
+      const choices = [{
+        value: lambertCoordinaat,
+        label: 'Lambert-coördinaat: ' + lambertCoordinaat.toString(),
+      }];
+
+      if (data && data.LocationResult) {
+        data.LocationResult.map((locationResult) => {
+          return {
+            value: locationResult,
+            label: locationResult.FormattedAddress,
+          };
+        }).forEach((choice) => {
+          choices.push(choice);
+        });
+      }
+      this._selectElement.choices = choices;
+    });
+  }
+
   _addChoiceEventListener() {
     if (!this.__choiceEventListenerRegistered) {
       this.__choiceEventListenerRegistered = true;
       this.__choiceEventListener = this._selectElement.addEventListener('choice', (event) => {
         if (event && event.detail && event.detail.choice) {
           const value = event.detail.choice.value;
 
-          // TODO stefanborghys: 28/01/21 prefer to use instanceof but could not import LambertCoordinaat in the test and replaced it for now
-          // instanceof LambertCoordinaat
-          if (value.x !== undefined && value.y !== undefined ) {
-            this._map.zoomToGeometry({
-              type: 'Point',
-              coordinates: [value.x, value.y],
-            }, 10);
+          if (this._isLambertCoördinaat(value)) {
```

Comment:
Tricky om UTF-8 (ö) te gaan gebruiken hier? Zien dat we geen enconding-prul krijgen ...?